File: app/models/post.rb
# == Schema Information
#
# Table name: posts
#
#  id         :integer          not null, primary key
#  like_count :integer          default(0)
#  user_id    :integer
#  created_at :datetime         not null
#  updated_at :datetime         not null
#

class Post < ActiveRecord::Base
  belongs_to :user
  has_many :likes

  has_many :song_posts, class_name: 'SongPost'
  # has_many :songs, through: :song_posts
  validates :user_id, presence: true
  validates :like_count, :numericality => {:greater_than_or_equal_to => 0}

  before_create :default_values
  def songs
    Song.joins("INNER JOIN song_posts ON songs.id = song_posts.song_id WHERE song_posts.post_id = #{self.id}").select("songs.*")
  end

  def as_json(options = {})
      more_hash = {
        like_count: self.like_count,
        is_liked: User.find(options[:id]).liked?(self.id)
      }
    super(options).merge(post: more_hash, song: self.songs.first, user: self.user.as_json(limited: true))
  end

  private
  def default_values
    self.like_count = 0
  end
end

File: src/app/tempo/models/post.py
from . import *

class Post(Base):
  __tablename__ = 'posts'

  id = db.Column(db.Integer, primary_key = True)
  like_count = db.Column(db.Integer, default = 0)
  user_id = db.Column(db.Integer, db.ForeignKey('users.id', ondelete = 'CASCADE'))
  views = db.Column(db.Integer, default = 0)

  user = db.relationship('User', cascade='all,delete')
  song_posts = db.relationship('SongPost', cascade='all,delete')
  likes = db.relationship('Like', cascade='all,delete')

  def __init__(self, **kwargs):
    self.like_count = kwargs.get('like_count', 0)
    self.user_id = kwargs.get('user_id', 0)
    self.views = kwargs.get('views', 0)
